From dfb8c93d4e9754e56ce805088e33fed2668c2834 Mon Sep 17 00:00:00 2001
From: Hood Chatham <roberthoodchatham@gmail.com>
Date: Mon, 19 Jun 2023 12:32:05 -0700
Subject: [PATCH 7/7] Remove trampoline definitions

We need to make our own JSPI aware versions. They live in src/core/trampoline.c.

---
 Objects/descrobject.c  | 15 +++++++--------
 Objects/methodobject.c |  3 ---
 Python/import.c        |  7 -------
 Python/importdl.h      |  4 ----
 4 files changed, 7 insertions(+), 22 deletions(-)

diff --git a/Objects/descrobject.c b/Objects/descrobject.c
index 4d8b83758b..5f172cf252 100644
--- a/Objects/descrobject.c
+++ b/Objects/descrobject.c
@@ -15,14 +15,13 @@ class property "propertyobject *" "&PyProperty_Type"
 
 // see pycore_object.h
 #if defined(__EMSCRIPTEN__) && defined(PY_CALL_TRAMPOLINE)
-#include <emscripten.h>
-EM_JS(int, descr_set_trampoline_call, (setter set, PyObject *obj, PyObject *value, void *closure), {
-    return wasmTable.get(set)(obj, value, closure);
-});
-
-EM_JS(PyObject*, descr_get_trampoline_call, (getter get, PyObject *obj, void *closure), {
-    return wasmTable.get(get)(obj, closure);
-});
+
+PyObject*
+descr_get_trampoline_call(getter get, PyObject *obj, void *closure);
+
+int
+descr_set_trampoline_call(setter set, PyObject *obj, PyObject *value, void *closure);
+
 #else
 #define descr_set_trampoline_call(set, obj, value, closure) \
     (set)((obj), (value), (closure))
diff --git a/Objects/methodobject.c b/Objects/methodobject.c
index 953cf4666d..969338d701 100644
--- a/Objects/methodobject.c
+++ b/Objects/methodobject.c
@@ -558,7 +558,4 @@ cfunction_call(PyObject *func, PyObject *args, PyObject *kwargs)
 #if defined(__EMSCRIPTEN__) && defined(PY_CALL_TRAMPOLINE)
 #include <emscripten.h>
 
-EM_JS(PyObject*, _PyCFunctionWithKeywords_TrampolineCall, (PyCFunctionWithKeywords func, PyObject *self, PyObject *args, PyObject *kw), {
-    return wasmTable.get(func)(self, args, kw);
-});
 #endif
diff --git a/Python/import.c b/Python/import.c
index 07a8b90092..f59d2b6bb8 100644
--- a/Python/import.c
+++ b/Python/import.c
@@ -958,13 +958,6 @@ PyImport_GetImporter(PyObject *path)
     return get_path_importer(tstate, path_importer_cache, path_hooks, path);
 }
 
-#if defined(__EMSCRIPTEN__) && defined(PY_CALL_TRAMPOLINE)
-#include <emscripten.h>
-EM_JS(PyObject*, _PyImport_InitFunc_TrampolineCall, (PyModInitFunction func), {
-    return wasmTable.get(func)();
-});
-#endif // __EMSCRIPTEN__ && PY_CALL_TRAMPOLINE
-
 static PyObject*
 create_builtin(PyThreadState *tstate, PyObject *name, PyObject *spec)
 {
diff --git a/Python/importdl.h b/Python/importdl.h
index 26d18b626d..6a3d5880c0 100644
--- a/Python/importdl.h
+++ b/Python/importdl.h
@@ -12,11 +12,7 @@ extern PyObject *_PyImport_LoadDynamicModuleWithSpec(PyObject *spec, FILE *);
 
 typedef PyObject *(*PyModInitFunction)(void);
 
-#if defined(__EMSCRIPTEN__) && defined(PY_CALL_TRAMPOLINE)
-extern PyObject *_PyImport_InitFunc_TrampolineCall(PyModInitFunction func);
-#else
 #define _PyImport_InitFunc_TrampolineCall(func) (func)()
-#endif
 
 /* Max length of module suffix searched for -- accommodates "module.slb" */
 #define MAXSUFFIXSIZE 12
-- 
2.25.1

